### DISCUSSION TITLE
How can I toggle class names on nodes?

### DISCUSSION THREAD

**Discussion by lumenwrites** [ueberdosis/tiptap#2248] [General Questions]
Hey guys! I'm new to TipTap, really loving it so far.

I need a bit of help. My goal is to create a button that toggles a class name on my `h1` elements.

I have successfully figured out how to add a class name to the element, as described [here](https://github.com/ueberdosis/tiptap/issues/586#issuecomment-836466931):
```
const CustomHeading = Heading.extend({
  addCommands() {
    return {
      ...this.parent?.(),
      toggleClassName: (className) => ({ commands }) => {
        return commands.updateAttributes('heading', { class: className })
      },
    }
  },
})
```

Now when I click on my button once, it runs `editor.chain().focus().toggleClassName('post-header').run()`, and the `post-header` class is successfully added to my `h1`.

But how can I toggle it off?

When I click on the button one more time, I want it to remove the `post-header` class.

Can you help me out?

**Comment by Notercom** [comment]
editor.isActive('paragraph', {class: 'post-header'})
